akka-projection-grpc/src/main/scala/akka/projection/grpc/consumer/GrpcQuerySettings.scala

```diff
@@ -4,25 +4,87 @@
 
 package akka.projection.grpc.consumer
 
+import akka.grpc.scaladsl.Metadata
+import akka.grpc.scaladsl.MetadataBuilder
 import com.typesafe.config.Config
 
+import java.util.Optional
+import scala.jdk.CollectionConverters.MapHasAsScala
+import scala.jdk.OptionConverters.RichOptional
+
 object GrpcQuerySettings {
-  def apply(config: Config): GrpcQuerySettings =
-    new GrpcQuerySettings(config)
+  def apply(config: Config): GrpcQuerySettings = {
+    val streamId = config.getString("stream-id")
+    require(
+      streamId != "",
+      "Configuration property [stream-id] must be an id exposed by the streaming side (but was empty).")
+
+    val protoClassMapping: Map[String, String] = {
+      import scala.jdk.CollectionConverters._
+      config.getConfig("proto-class-mapping").root.unwrapped.asScala.toMap.map {
+        case (k, v) => k -> v.toString
+      }
+
+    }
+
+    val additionalHeaders: Option[Metadata] = {
+      import scala.jdk.CollectionConverters._
```

Comment:
Scala 2.12? I see that we use `import akka.util.ccompat.JavaConverters._` in some other places in Akka Projections.

Reply:
Ups. That should fail validation then. We probably didn't set the default/main Scala version to 2.12 so IntelliJ picks up what's reasonable for 2.13, I'll bump that and fix